django/thunderstore/api/cyberstorm/views/team.py

```diff
@@ -0,0 +1,70 @@
+from django.db.models import Q, QuerySet
+from rest_framework.exceptions import NotAuthenticated, NotFound, PermissionDenied
+from rest_framework.filters import OrderingFilter
+from rest_framework.generics import ListAPIView, RetrieveAPIView
+from rest_framework.permissions import BasePermission
+
+from thunderstore.account.models.service_account import ServiceAccount
+from thunderstore.api.cyberstorm.serializers import (
+    CyberstormServiceAccountSerializer,
+    CyberstormTeamMemberSerializer,
+    CyberstormTeamSerializer,
+)
+from thunderstore.api.utils import CyberstormAutoSchemaMixin
+from thunderstore.repository.models.team import Team, TeamMember
+
+
+class TeamDetailAPIView(CyberstormAutoSchemaMixin, RetrieveAPIView):
+    permission_classes = []
+    serializer_class = CyberstormTeamSerializer
+    queryset = Team.objects.exclude(is_active=False)
+    lookup_field = "name__iexact"
+    lookup_url_kwarg = "team_id"
+
+
+class TeamMembershipPermission(BasePermission):
+    """
+    Ensure the user is a member of the Team.
+    """
+
+    def has_permission(self, request, view):
+        if not request.user.is_authenticated:
+            raise NotAuthenticated()
+
+        try:
+            team = Team.objects.exclude(is_active=False).get(
+                name__iexact=view.kwargs["team_id"],
+            )
+        except Team.DoesNotExist:
+            raise NotFound()
+
+        if not team.can_user_access(request.user):
+            raise PermissionDenied()
+
+        return True
```

Comment:
Changed to use a view. I'm not sure if the approach is much better. I now override the `check_permissions`, which in the vanilla implementation just loops through the `permission_classes`, so now this non-obviously makes the `permission_classes` class attribute to not tell the whole truth. Just as awkward but in a different way?

Also this might full well be due to not being familiar enough with DRF. There seems to be some sort of disconnect if we want to check the user has permissions to the team, on a list view that returns non-team items. E.g. seems `has_object_permission` isn't called at all, and `has_permission` also had some issue but I can't remember what it was anymore.

Reply:
> so now this non-obviously makes the permission_classes class attribute to not tell the whole truth. Just as awkward but in a different way?

Yeah, but I think it's never the right assumption to except `permission_classes` to be the _ONLY_ part of the view logic that handles permissions, so it's a bit of a false premise

As for object level permissions, I haven't seen an implementation of those in either Django or DRF that I'd prefer over a custom solution, but I haven't explored too deep either. The main problem usually is that I'd want to make permission logic as hard to get accidentally wrong as possible but also easy to see what's going on, which is not easy when you rely on frameworks for them.